packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx

```diff
@@ -97,6 +110,55 @@ export const AddEditExperimentAssignmentQueryModal: FC<EditExperimentAssignmentQ
     return null;
   }
 
+  const handleTestQuery = async () => {
+    setTestQueryResults(null);
+    setQueryError(null);
+
+    const value: ExposureQuery = {
+      name: exposureQuery.name,
+      query: userEnteredQuery,
+      id: dataSource.id,
+      userIdType: userEnteredUserIdType,
+      dimensions: userEnteredDimensions,
+      hasNameCol: userEnteredHasNameCol ? userEnteredHasNameCol : false,
+    };
+
+    try {
+      const requiredColumns = getRequiredColumns(value);
+      validateSQL(value.query, requiredColumns);
+
+      const res: TestQueryResults = await apiCall("/query/test", {
+        method: "POST",
+        body: JSON.stringify({
+          query: value.query,
+          datasourceId: value.id,
+          requiredColumns,
+        }),
+      });
+
+      if (res.includesNamedColumns && !value.hasNameCol) {
+        // If the query includes named columns, but hasNameCol is false
+        // force hasNameCol to true in order to identify if any required name columns
+        // are missing, enable hasNameCols for the user, and then throw applicable error.
+        value.hasNameCol = true;
+
+        const requiredColumns = getRequiredColumns(value);
+
+        form.setValue("hasNameCol", true);
+        validateSQL(value.query, requiredColumns);
```

Comment:
I don't think we want to revalidate here.  Just set `hasnameCol` to true and continue.  I made a comment in the back-end to only set `includesNamedColumns` if both of the required columns are defined.